incubator/virtualcluster/cmd/kubectl-vc/use.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/multi-tenancy/incubator/virtualcluster/pkg/syncer/conversion"
+)
+
+const vcKubeconfigHome = ".kube/vc/"
+
+var useCommand = cli.Command{
+	Name:      "use",
+	Usage:     "Switch kubeconfig among virtualclusters",
+	ArgsUsage: "VC_NAME",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "namespace, n",
+			Usage: "If present, the namespace scope for this CLI request, or use 'default'",
+		},
+		cli.BoolFlag{
+			Name:  "interactive, i",
+			Usage: "Launch a shell and switch to the specific virtualcluster kubeconfig",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		ns := metav1.NamespaceDefault
+		if fns := cctx.String("namespace"); fns != "" {
+			ns = fns
+		}
+
+		name := cctx.Args().Get(0)
+		if name == "" {
+			return errors.New("virtualcluster name is required")
+		}
+
+		kbFilePath, err := placeVCKubeconfig(ns, name)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("kubeconfig for virtualcluster %s/%s is placed at:\n\n\t%s\n\n", ns, name, kbFilePath)
+
+		if cctx.Bool("interactive") {
```

Comment:
In my opinion, the -i command is mandatory. There is not much of use case people just want to generate the kubeconfig and don't want to operate on it. 

I would suggest to always call enterVCShell and again remove one input parameter and make the tool easier to use. 

My point is that if 99% of time people need to input a bool parameter which is true, that bool behavior should be added to code by default.